app/views/layouts/base.html.erb

```diff
@@ -32,27 +32,24 @@
       <% end %>
     </script>
     <%= javascript_include_tag "locale/#{FastGettext.locale}/app" %>
-    <%= javascript_include_tag *webpack_asset_paths('foreman-vendor', :extension => 'js') %>
-    <%= javascript_include_tag *webpack_asset_paths('vendor', :extension => 'js') %>
-    <%= javascript_include_tag *webpack_asset_paths('bundle', :extension => 'js') %>
-    <%= javascript_include_tag 'application' %>
-    <%= webpacked_plugins_with_global_js %>
-    <%= webpack_dev_server %>
-    <%= yield(:javascripts) %>
-
-    <script type="text/javascript">
-      <%= yield(:inline_javascripts) %>
-    </script>
 
     <%= yield(:head) %>
   </head>
 
   <body class="pf-m-redhat-font">
-    <!--[if lte IE 9]>
-      <%= _('You are using an unsupported browser.') %>
-    <![endif]-->
+   
     <% if User.current %>
       <% cache(TopbarSweeper.fragment_name, :skip_digest => true) do %>
```

Comment:
When adding it to the cache it seemed like the layout was rendered faster and there was a smoother transition between pages and components loading, but I think it's not required and scripts can also live in the body without the cache.

I was putting it here since the `<%= mount_react_app %>` was there

if you want, I could compare the performance with/out the cache

Reply:
I was more thinking about cache invalidation etc. It probably doesn't matter here, but if you call the same fragment from multiple pages it may end up with invalid caches.

Reply:
I do suspect it can cause problems because the `fragment_name` is defined as:
https://github.com/theforeman/foreman/blob/2dc52832034fb8dc289819ebc04f26af13902d33/app/services/topbar_sweeper.rb#L12-L14

So this is cached for each user. Then `mount_react_app` is defined as:

https://github.com/theforeman/foreman/blob/2dc52832034fb8dc289819ebc04f26af13902d33/app/helpers/layout_helper.rb#L2-L8

All of that data is not controller dependent, so it's fine.

However, I am worried that the other content could be problematic. In particular `javascripts` and `inline_javascripts` which are yielded. If we include controller specific JS, such as `app/assets/javascripts/hosts.js` on the hosts page then I'd think it can get cached across multiple pages thus making it can become unreliable. The order in which you navigate pages will determine if it works or not.

Another thing is cache efficiency. None of the things you include here are not user dependent. So including it in a per-user cache will mean you transfer more data in & out of the cache. You take up more cache space, meaning that under pressure the cache is more likely to evict data.

I don't know about the cost of calculating those JS tags, but I'm willing to bet that it's very cheap so caching it doesn't bring any performance benefit and perhaps even bring a net-negative.

If we find that it is expensive to calculate, then it deserves its own cache entry. One which is not bound to a specific user.

In conclusion: please move it out of the cached part.

Reply:
Thanks for the investigation, moved them out of the cache block,
and the pages seem to load faster than before.